zipkin-server/src/test/java/zipkin2/server/internal/prometheus/ITZipkinMetricsDirty.java

```diff
@@ -41,8 +39,14 @@
 import static zipkin2.server.internal.ITZipkinServer.url;
 
 /**
- * We cannot clear the micrometer registry easily, so we have recreate the spring context. This is
- * extremely slow, so please only add tests that require isolation here.
+ * Tests here look at values based on counter values, so need to run independently. It would seem
+ * correct to {@link PrometheusMeterRegistry#clear() clear the registry} to isolate counters
+ * incremented in one test from interfering with another. However, this clears the metrics
+ * themselves, resulting in an empty {@link PrometheusMeterRegistry#scrape()}.
```

Comment:
This not working as hoped is unfortunate indeed. Increments to a `Counter` reference held that is registered to a meter registry that is `clear`ed effectively do nothing (from the view of the registry) since the counter is no longer registered to the `MeterRegistry`. Making a new Spring context will make a new registry and all the metrics will be registered to that again in the remade beans. If using the API that retrieves or creates a meter, this isn't an issue (e.g. `registry.counter("my.counter").increment()`) because the first time after `clear` it will create and register a Counter. Perhaps we can improve this situation or at least document it better in Micrometer.

Reply:
if "reset" lingo in prom matches setting back to zero, it could avail another method name if the feature ever existed...